src/GUI/NUnitTestCore/HumanizerPluralizerTest.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using NUnit.Framework;
+using RevEng.Core;
+
+namespace UnitTests
+{
+    [TestFixture]
+    public class HumanizerPluralizerTest
+    {
+        [Test]
+        public void NotPluralizeAWordWhenExcluded()
+        {
+            HumanizerPluralizer.SetWordsNotToAlter(new List<string> { "Dog" });
+            var sut = new HumanizerPluralizer();
+            var result = sut.Pluralize("Dog");
+
+            Assert.True("Dog" == result);
+        }
+
+        [Test]
+        public void PluralizeAWordWhenNotExcluded()
+        {
+            HumanizerPluralizer.SetWordsNotToAlter(new List<string> { "Cat" });
+            var sut = new HumanizerPluralizer();
+            var result = sut.Pluralize("Dog");
+
+            Assert.True("Dogs" == result);
+        }
+
+        [Test]
+        public void NotPluralizeUserData()
+        {
+            HumanizerPluralizer.SetWordsNotToAlter(new List<string> { "UserData" });
+            var sut = new HumanizerPluralizer();
+            var result = sut.Pluralize("UserData");
+
+            Assert.True("UserData" == result);
+        }
+
+        [Test]
+        public void NotSingularizeUserData()
```

Comment:
Maybe add a not pluralize test also?

Reply:
I've restructured the tests to be more logical and cover more test cases.